### PR TITLE
[BEAM-1866] Add GAUGE_DATA case to metricUpdatesFromProto

### PR DESCRIPTION
**Pull request by lostluck**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/projects/BEAM/issues/) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue.
 - [ ] Write a pull request description that is detailed enough to understand:
   - [ ] What the pull request does
   - [ ] Why it does it
   - [ ] How it does it
   - [ ] Why this approach
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

**Comment by lostluck** [comment]
R: @kennknowles

**Comment by lostluck** [comment]
Nope, I don't mind. Could you provide me with the correct JIRA ticket for the counters work? I couldn't find it.

**Comment by lostluck** [comment]
@kennknowles PTAL. Is there a JIRA ticket I should refer to?

**Comment by kennknowles** [comment]
Looks like it is BEAM-1866

**Comment by kennknowles** [comment]
LGTM and that test failure is now sickbayed. I'll kick it.

**Comment by kennknowles** [comment]
retest this please

File: runners/core-java/src/test/java/org/apache/beam/runners/core/metrics/MetricsTranslationTest.java [hunk 1]
```diff
@@ -38,6 +38,7 @@
   // Transform names are arbitrary user-meaningful steps in processing
   private static final String TRANSFORM1 = "transform1";
   private static final String TRANSFORM2 = "transform2";
+  private static final String TRANSFORM2 = "transform3";
```
Comment (kennknowles) [inline runners/core-java/src/test/java/org/apache/beam/runners/core/metrics/MetricsTranslationTest.java]: Typo - `TRANSFORM3`.

**Comment by kennknowles** [comment]
LGTM modulo minor typos that I've fixed in a derived PR. I'll go ahead and merge with fixups after getting green.

**Pull request closed without merging**